samples: mesh/onoff_level_lighting_vnd_app: States binding corrections

vikrant8052 opened the pull request:
Make corrections in state binding of Servers as per Mesh Model
Specifications. Previously, when OnPowerUp state equal to 0x02,
then Light Lightness actual state was not assign to last power down value.
Plus when Generic OnOff state changes by client, then Light Lightness Actual
state value get assigned as Light Lightness Last value instead of default one.
All these issues has fixed here.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8051 commented:
@carlescufi @jhedberg  After several testing I found that current demo on main branch is not exactly following Mesh Model specification in some conditions. Excuse me for that !!

I've rectified those things so please approved this as early as possible.

jhedberg reviewed (changes requested):
Thanks for fixing these issues, however would it be possible to get a more detailed description of what exactly is being fixed (in the commit message)? Also, if these are independent fixes, please split them up into multiple patches. It's quite easy to do using e.g. `git add -i` to add only specific chunks to certain patches.

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.h:
```diff
@@ -10,11 +10,12 @@
-enum lightness { ONOFF = 0x01, LEVEL, ACTUAL, LINEAR, CTL, IGNORE = 0xFF };
-enum temperature { ONOFF_t = 0x01, LEVEL_t, CTL_t, IGNORE_t = 0xFF };
+enum lightness { ONPOWERUP = 0x01, ONOFF, LEVEL, ACTUAL, LINEAR, CTL, IGNORE};
+enum temperature { ONOFF_t = 0x01, LEVEL_t, CTL_t, IGNORE_t};
```
Unrelated to this patch since it already slipped through in the original code, however could you rename the _t suffix to something else? That's normally used for typedefs in Zephyr which makes it quite confusing. Also, keep the names all capital letters. Perhaps `LEVEL_TEMP` or something like that?

vikrant8051 replied:
I will change name to ONOFF_TEMP. And it is related to state binding. 

I will add details in descriptions.

vikrant8051 replied:
Now changed 
enum temperature { ONOFF_t = 0x01, LEVEL_t, CTL_t, IGNORE_t}; to
enum temperature { ONOFF_TEMP = 0x01, LEVEL_TEMP, CTL_TEMP, IGNORE_TEMP};

vikrant8052 committed:
samples: mesh/onoff_level_lighting_vnd_app: States binding corrections

Make corrections in state binding of Servers as per Mesh Model
Specifications. Previously, when OnPowerUp state equal to 0x02,
then Light Lightness actual state was not assign to last power down value.
Plus when Generic OnOff state changes by client, then Light Lightness Actual
state value get assigned as Light Lightness Last value instead of default one.
All these issues has fixed here.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

jhedberg commented:
@vikrant8051 thanks, the commit message is now much better! I still need to stress the fact that it's good practice to split independent changes into separate patches. You've now made the situation worse by baking in the _t -> _TEMP suffix change into this same patch, even though that's clearly a separate change. It can still be in the same pull request, but it should really be a separate patch. This might be a good opportunity to learn/practice the use of `git add -i`. You'd start off with `git reset HEAD~1`, and do `git add -i; git commit -v -s` for each individual patch you want to create, until there are no more unstaged changes.

Following this principle makes it easier to track the change history, e.g. mapping the changes to code to a descriptive commit message of those changes. It also helps find the cause of bugs later on with the help of `git bisect`. Bisecting will often quickly find the offending commit, but if there's lots of unrelated changes in that commit it can be hard to pinpoint what exactly caused the breakage.

vikrant8051 commented:
@jhedberg  Sorry for that. Please excuse me for this time or for this PR. 
I'm now sure that it is completely Bug free. In upcoming days, I will improve my git related skills.

vikrant8051 commented:
.
